### PR TITLE
Save latests schema

### PR DESCRIPTION
**Pull request by nechaido**

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -378,11 +379,18 @@ class Application extends api.events.EventEmitter {
   //   callback <Function>
   //     err <Error> | <null>
   updateSchemas({ version }, callback) {
+    if (this.latestSchema && version < this.latestSchema.version) {
```
Comment (SemenchenkoVitaliy) [inline lib/application.js]: I think there should also be check if `version === this.latestSchema.version` so it won't load the same schema again.

File: lib/application.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ class Application extends api.events.EventEmitter {
     this.config = new impress.Config(this);
     this.scheduler = new impress.Scheduler(this);
     this.schemas = new Map(); // Map of application schemas in metaschema format
+    this.latestSchema = null;
```
Comment (tshemsedinov) [inline lib/application.js]: Why do we have both `latestSchema` and `schemas` fields?

Comment (nechaido) [reply lib/application.js]: So that we can easily determine the latest available schema.

File: lib/application.js [hunk 3]
```diff
@@ -378,11 +379,24 @@ class Application extends api.events.EventEmitter {
   //   callback <Function>
   //     err <Error> | <null>
   updateSchemas({ version }, callback) {
+    if (this.latestSchema) {
+      if (version === this.latestSchema.version) {
+        callback(null);
+        return;
+      }
+      if (version < this.latestSchema.version) {
+        application.log.warning('Attempt to update schema to a lower version');
```
Comment (lundibundi) [inline lib/application.js]:
```suggestion
        application.log.warning('Attempt to update schema to a lower version. Ignored');
```
?

**Comment by tshemsedinov** [comment]
Landed in e7f8f28

**Pull request closed without merging**